core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -274,6 +295,13 @@ private void maybeLogPathDetails() throws IOException {
         }
     }
 
+    private void maybeLogHeapDetails() {
+        ByteSizeValue maxHeapSize = JvmInfo.jvmInfo().getMem().getHeapMax();
+        Boolean usingCompressedOops = JvmInfo.jvmInfo().usingCompressedOops();
+        String usingCompressedOopsStatus = usingCompressedOops == null ? "unknown" : Boolean.toString(usingCompressedOops);
+        logger.info("heap size [{}], compressed ordinary object pointers [{}]", maxHeapSize, usingCompressedOopsStatus);
```

Comment:
> Sorry I am late, but I think this should go under debug logging, as by default, this will be logged each time a node starts, and I don't think it is important enough to log each time a node starts...

@kimchy We log mount points, disk space metrics, whether or not the disks are maybe spinning and the underlying file system types at the info level; I opted for consistency with that.